Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/split/CheckInSplitter.kt

```diff
@@ -0,0 +1,94 @@
+package de.rki.coronawarnapp.eventregistration.checkins.split
+
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.seconds
+import org.joda.time.Instant
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+import kotlin.math.ceil
+import kotlin.math.max
+
+/**
+ * Splits a [CheckIn] by midnight UTC time into multiple [CheckIn]s across multiple days.
+ *
+ * @return [List] of [CheckIn]s
+ */
+fun CheckIn.splitByMidnightUTC(): List<CheckIn> {
+    val startTimeSeconds = checkInStart.seconds
+    val endTimeSeconds = checkInEnd?.seconds ?: 0L
```

Comment:
How about an early return if the checkInEnd is null or before the checkInStart?

Reply:
No,  I would like to keep that way.